book/src/specification/logical.md

```diff
@@ -1,3 +1,788 @@
+   > of the specification. It arises for instance for nested lists, which may
+   > result in something of the form
+   > \\(\textrm{Stream}(d=1, T_e=\textrm{Stream}(d=1, s=\textrm{Sync}, T_e=...))\\).
+   > The outer stream is not necessary here; all the dimensionality information
+   > is carried by the inner stream.
+   >
+   > It could be argued that the above example should be reduced to
+   > \\(\textrm{Stream}(d=2, s=\textrm{Sync}, T_e=...)\\) by said external
+   > tool. Indeed, this description is equivalent. This reduction is however
+   > very difficult to define or specify in a way that it handles all cases
+   > intuitively, hence this more pragmatic solution was chosen.
+
+> As the \\(\textrm{Stream}\\) node carries many parameters and can therefore
+> be hard to read, some abbreviations are in order:
+> 
+>  - \\(\textrm{Dim}(T_e, n, c, T_u) \ \rightarrow\ \textrm{Stream}(T_e, n, 1, \textrm{Sync},    c, \textrm{Forward}, T_u, \textrm{false})\\)
+>  - \\(\textrm{New}(T_e, n, c, T_u) \ \rightarrow\ \textrm{Stream}(T_e, n, 0, \textrm{Sync},    c, \textrm{Forward}, T_u, \textrm{false})\\)
+>  - \\(\textrm{Des}(T_e, n, c, T_u) \ \rightarrow\ \textrm{Stream}(T_e, n, 0, \textrm{Desync},  c, \textrm{Forward}, T_u, \textrm{false})\\)
+>  - \\(\textrm{Flat}(T_e, n, c, T_u)\ \rightarrow\ \textrm{Stream}(T_e, n, 0, \textrm{Flatten}, c, \textrm{Forward}, T_u, \textrm{false})\\)
+>  - \\(\textrm{Rev}(T_e, n, c, T_u) \ \rightarrow\ \textrm{Stream}(T_e, n, 0, \textrm{Sync},    c, \textrm{Reverse}, T_u, \textrm{false})\\)
+> 
+> For the above abbreviations, the following defaults apply in addition:
+> 
+>  - \\(T_u = \textrm{Null}\\);
+>  - \\(c =\\) complexity of parent stream (cannot be omitted if there is no
+>    parent);
+>  - \\(n = 1\\).
+
+#### Null
+
+The \\(\textrm{Null}\\) leaf node indicates the transferrence of one-valued
+data: it is only valid value is \\(\varnothing\\) (null).
+
+> This is useful primarily when used within a \\(\textrm{Union}\\), where the
+> data type itself carries information. Furthermore, \\(\textrm{Stream}\\)
+> nodes carrying a \\(\textrm{Null}\\) will always result in a physical stream
+> (whereas \\(\textrm{Stream}\\) nodes carrying only other streams will be
+> "optimized out"), which may still carry dimensionality information, user
+> data, or even just transfer timing information.
+
+#### Bits
+
+The \\(\textrm{Bits}\\) leaf node, defined as \\(\textrm{Bits}(b)\\), indicates
+the transferrence of \\(2^b\\)-valued data carried by means of a group of
+\\(b\\) bits, where \\(b\\) is a positive integer.
+
+> Tydi does not specify how the bits should be interpreted: such interpretation
+> is explicitly out of scope. Where necessary for examples, this specification
+> will usually use unsigned binary numbers or ASCII characters expressed using
+> 8 bits.
+
+### Group
+
+The \\(\textrm{Group}\\) node acts as a product type for composition. It is
```

Comment:
remove "for composition" since this implied by being a product type

Reply:
Fixed in ec8d1c90